components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/WebNotificationIntentProcessor.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.webnotifications
+
+import android.content.Intent
+import mozilla.components.feature.intent.processing.IntentProcessor
+import org.mozilla.geckoview.WebNotification as GeckoViewWebNotification
+
+/**
+ * Intent processor that tries matching a web notification and delegating a click interaction with it.
+ */
+class WebNotificationIntentProcessor : IntentProcessor {
+    /**
+     * Processes an incoming intent expected to contain information about a web notification.
+     * If such information is available this will inform the web notification about it being clicked.
+     */
+    override fun process(intent: Intent): Boolean {
+        @Suppress("MoveVariableDeclarationIntoWhen")
+        val geckoNotification =
+            (intent.extras?.get(NativeNotificationBridge.EXTRA_ON_CLICK) as? GeckoViewWebNotification)
```

Comment:
This is a problem here as the `feature` components shouldn't directly depend on Gecko. At least this breaks the current design where we make sure they only depend on `concept-engine`. 

Likely the easiest way to fix this is by adding a method to the `engine` interface e.g., `handleWebNotificationClick(intent)` and the move this code into the implementation of it in `GeckoEngine`.

Reply:
And pass the engine to this new intent processor to call `handleWebNotificationClick`.

Reply:
This does look much cleaner. Thank you!